specification.md

```diff
@@ -1,15 +1,279 @@
+  and writes the policy (in-toto layout) that indicates how different steps 
+  of the software supply chain interrelate.  This party also serves as the
+  SBOMit root-of-trust.
+
+* a (possibly empty) set of in-toto sub-layout creators.  These are identical
+  in action to the in-toto layout creator, but they are only trusted for the
+  subset of the layout which the layout creator delegates.  
+
+
+
+We are able to limit attacker damage and/or trace the location of the 
+compromise(s) in cases where:
+* an attacker may possess cryptographic keys for any of the functionaries (the
+  parties performing the software supply chain steps) in the system.  
+  **Impact:** Prevents modification of items not allowed by the in-toto layout.
+  Provides traceability in all cases.  No impact without the ability to 
+  interject this metadata into the supply chain.
+* an attacker may tamper with one or more steps of the software supply chain.
+  For example, the build process, testing, packaging, etc.  
+  **Impact:** Identical to the prior case.
+* an attacker obtains a sub-layout key.  Note that this also requires the 
+  ability to inject sublayout metadata into the supply chain such that other
+  parties include it.  **Impact:** The impact could be prevented depending on 
+  the restrictions placed upon the delegated sub-layout.  However, traceability
+  always exists.  This is also similar to a functionary compromise.
+* an attacker may become a man-in-the-middle between any steps of the 
+  system.  **Impact:** Without further capabilities, this has no impact.
+* an attacker may possess the key used to sign the SBOM resulting from the
+  SBOMit process.  **Impact:** Prevention for parties performing full SBOMit
+  verification.  Traceability otherwise.
+* an attacker may compromise an SBOM mutator key or an SBOM mutator may act
+  maliciously.  **Impact:** Traceability in all cases.  Depending on the 
+  changes made to the SBOM, this may be detected by review of the resulting
+  SBOM.  Changes that override in-toto derived information are specifically 
+  flagged and unlikely to be accepted. XXXXXX
+* an attacker is able to compromise a SIT generator key or a SIT generator 
+  directly. **Impact:** Protection for clients who obtain the SBOMit document.
+  Traceability for other clients. Note that the SIT will contain a refernce to
+  the SBOMit document, but this also may be modified in this case.  However,
+  the client should have the in-toto layout key and so will notice this 
+  action if retrieving the SBOMit document from the the SIT.
+* an attacker is able to compromise the in-toto layout key, which serves as
+  the root-of-trust for the system.  **Impact:** Traceability.  Later analysis
+  can show this was the cause, but users will trust a new, maliciously 
+  generated layout which replaces signing keys.
+
+From (Endor Lab's top 10 OSS risks)[https://www.endorlabs.com/blog/introducing-the-top-10-open-source-software-oss-risks], 
```

Comment:
Maybe this should come much earlier. I really don't understand what this threat model is trying to say. What sort of attacker goals are we most worried about? Those tampering with just the list of dependencies, or does the tampering extend to the software artefacts themselves?

Reply:
This covers both cases.  I don't want to describe someone else's work here, but wanted to give some pointers to an external list to show we're handling a broad set of cases.